File: src/main/java/com/noximity/remmyChat/models/ChatUser.java
package com.noximity.remmyChat.models;

import java.util.UUID;

public class ChatUser {

    private final UUID uuid;
    private String currentChannel;
    private UUID lastMessagedPlayer;
    private boolean msgToggle;
    private boolean socialSpy;

    public ChatUser(UUID uuid, String defaultChannel) {
        this.uuid = uuid;
        this.currentChannel = defaultChannel;
        this.msgToggle = true;
        this.socialSpy = false;
    }

    public ChatUser(UUID uuid, String defaultChannel, boolean msgToggle, boolean socialSpy) {
        this.uuid = uuid;
        this.currentChannel = defaultChannel;
        this.msgToggle = msgToggle;
        this.socialSpy = socialSpy;
    }

    public UUID getUuid() {
        return uuid;
    }

    public String getCurrentChannel() {
        return currentChannel;
    }

    public void setCurrentChannel(String currentChannel) {
        this.currentChannel = currentChannel;
    }

    public UUID getLastMessagedPlayer() {
        return lastMessagedPlayer;
    }

    public void setLastMessagedPlayer(UUID lastMessagedPlayer) {
        this.lastMessagedPlayer = lastMessagedPlayer;
    }

    public boolean isMsgToggle() {
        return msgToggle;
    }

    public void setMsgToggle(boolean msgToggle) {
        this.msgToggle = msgToggle;
    }

    public boolean isSocialSpy() {
        return socialSpy;
    }

    public void setSocialSpy(boolean socialSpy) {
        this.socialSpy = socialSpy;
    }
}

File: bin/main/com/noximity/remmyChat/models/ChatUser.kt
package com.noximity.remmyChat.models

import java.util.*

class ChatUser {
    val uuid: UUID
    var currentChannel: String
    var lastMessagedPlayer: UUID? = null
    var lastMessageSender: UUID? = null
    var isMsgToggle: Boolean
    var isSocialSpy: Boolean
    var isMessagesEnabled: Boolean
        get() = isMsgToggle
        set(value) { isMsgToggle = value }

    // Track last message times per channel
    private val lastMessageTimes = mutableMapOf<String, Long>()

    constructor(uuid: UUID, defaultChannel: String) {
        this.uuid = uuid
        this.currentChannel = defaultChannel
        this.isMsgToggle = true
        this.isSocialSpy = false
    }

    constructor(uuid: UUID, defaultChannel: String, msgToggle: Boolean, socialSpy: Boolean) {
        this.uuid = uuid
        this.currentChannel = defaultChannel
        this.isMsgToggle = msgToggle
        this.isSocialSpy = socialSpy
    }


    /**
     * Get the last message time for a specific channel
     */
    fun getLastMessageTime(channelName: String): Long {
        return lastMessageTimes[channelName] ?: 0L
    }

    /**
     * Update the last message time for a specific channel
     */
    fun updateLastMessageTime(channelName: String, time: Long = System.currentTimeMillis()) {
        lastMessageTimes[channelName] = time
    }

    /**
     * Clear message times for all channels
     */
    fun clearMessageTimes() {
        lastMessageTimes.clear()
    }
}
